commit 575dd729cb434c7a99074d2eecf5ee4b0356102a
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-08-01T21:55:17+00:00 (10 weeks ago)
    
    TRITON-1841 Updating NICs in VMAPI should call out to NAPI where appropriate

diff --git a/lib/workflows/update-nics.js b/lib/workflows/update-nics.js
index a89b3b5..49cdb73 100644
--- a/lib/workflows/update-nics.js
+++ b/lib/workflows/update-nics.js
@@ -12,7 +12,7 @@
  * A brief overview of this source file: what is its purpose.
  */
 
-var async = require('async');
+var vasync = require('vasync');
 var common = require('./job-common');
 var VERSION = '7.0.4';
 
@@ -38,6 +38,33 @@ function setupRequest(job, cb) {
 }
 
 
+/*
+ * This handles updating the NICs in NAPI. It's important to do this before
+ * updating the VM on the Compute Node so that net-agent has a chance to see
+ * the values, and won't try reverting them back to match NAPI.
+ */
+function updateNics(job, cb) {
+    var napi = new sdcClients.NAPI({
+        url: napiUrl,
+        headers: { 'x-request-id': job.params['x-request-id'] }
+    });
+
+    function updateNic(nic, done) {
+        napi.updateNic(nic.mac, nic, done);
+    }
+
+    vasync.forEachPipeline({
+        inputs: job.params.update_nics,
+        func: updateNic
+    }, function (err) {
+        if (err) {
+            cb(err);
+            return;
+        }
+
+        cb(null, 'Updated NAPI NICs');
+    });
+}
 
 var workflow = module.exports = {
     name: 'update-nics-' + VERSION,
@@ -66,6 +93,16 @@ var workflow = module.exports = {
         retry: 1,
         body: common.waitOnVMTicket,
         modules: { sdcClients: 'sdc-clients' }
+    }, {
+        name: 'napi.update_nics',
+        timeout: 10,
+        retry: 1,
+        body: updateNics,
+        modules: {
+            restify: 'restify',
+            sdcClients: 'sdc-clients',
+            vasync: 'vasync'
+        }
     }, {
         name: 'cnapi.update_vm',
         timeout: 10,
